google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiManagementPanel.java

```diff
@@ -0,0 +1,278 @@
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.event.TableModelListener;
+import javax.swing.table.DefaultTableCellRenderer;
+import javax.swing.table.TableModel;
+
+/**
+ * Panel allowing management activity of cloud APIs including api enablement and service account
+ * creation.
+ */
+public class GoogleCloudApiManagementPanel {
+
+  private JPanel panel;
+  private ProjectSelector projectSelector;
+  private JLabel selectProjectLabel;
+  private JBScrollPane apisScrollPane;
+  private JPanel apisPanel;
+  private JPanel selectedApisPanel;
+  private JPanel selectedApisWrapper;
+  private JTable selectedApisTable;
+  private JPanel allApisPanel;
+  private JPanel allApisWrapper;
+  private JTable allApisTable;
+
+  private static final int MIN_CLOUD_API_COL_WIDTH = 300;
+  private static final int MIN_CLOUD_API_ENABLEMENT_COL_WIDTH = 100;
+  private static final int CLOUD_API_NAME_COL = 0;
+  private static final int CLOUD_API_ENABLEMENT_COL = 1;
+
+  GoogleCloudApiManagementPanel() {
+    apisPanel.setBorder(BorderFactory.createEmptyBorder());
+
+    selectedApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    selectedApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.selectedapis.title")));
+
+    allApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    allApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.allapis.title")));
+
+    if (projectSelector.getSelectedProject() != null) {
+      showApiEnablementUi();
+    }
+    projectSelector.addProjectSelectionListener(project -> showApiEnablementUi());
+  }
+
+  private void showApiEnablementUi() {
+    apisScrollPane.setVisible(true);
+    selectProjectLabel.setVisible(false);
+  }
+
+  /** Initializes the table models of the API tables based on the user's library selection. */
+  // TODO(eshaul) its very important that we add an equals method to CloudLibrary so that the set
+  // operation is reliable
+  void init(List<CloudLibrary> allLibraries, Set<CloudLibrary> selectedLibraries) {
+    selectedApisTable.setModel(new ApisTableModel(selectedLibraries));
+    selectedApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_NAME_COL)
+        .setMinWidth(MIN_CLOUD_API_COL_WIDTH);
+    selectedApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_ENABLEMENT_COL)
+        .setMinWidth(MIN_CLOUD_API_ENABLEMENT_COL_WIDTH);
+
+    allApisTable.setModel(
+        new ApisTableModel(Sets.difference(Sets.newHashSet(allLibraries), selectedLibraries)));
+    allApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_NAME_COL)
+        .setMinWidth(MIN_CLOUD_API_COL_WIDTH);
+    allApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_ENABLEMENT_COL)
+        .setMinWidth(MIN_CLOUD_API_ENABLEMENT_COL_WIDTH);
+  }
+
+  JPanel getPanel() {
+    return panel;
```

Comment:
`panel` as a variable and getter sound a bit too generic? Maybe name it after the class itself meaning it contains everything in it - i.e. `googleCloudApiManagementPanel` or `contentPanel`?

Reply:
sure. I'll change it to contentPanel